scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/JobState.java

```diff
@@ -193,6 +194,104 @@ public JobPriority getPriority() {
      */
     public abstract Map<TaskId, TaskState> getHMTasks();
 
+
+    /**
+     * To get the task as an array list and filtered by a given tag.
+     * @param tag the used to filter the tasks.
+     * @return the set of filtered task states.
+     */
+    public List<TaskState> getTaskByTag(final String tag){
+        List<TaskState> tasks = this.getTasks();
+        return (List<TaskState>) CollectionUtils.select(tasks, new Predicate() {
```

Comment:
GetTaskByTag is in JobState to avoid repeat the same logic in subclasses. But classes are stocked only in the subclasses so I have to call the abstract method getTasks. In the case of JobStateImpl, getTask take all tasks one by one to extract TaskState from TaskStateData and return the resulting list. So in getTaskByTag in JobState, I filter after and do a lot of operations for nothing. For optimization purposes, I overrided in JobStateImpl to extract after the filtering. 

Anyway, with the refactoring to use SchedulerDBManager, this repeatition will be solved.